client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -155,13 +216,20 @@ class RequiredPluginsInstallView extends Component {
 			}
 		}
 
+		let stepIndex = this.state.stepIndex;
+
+		if ( signupIsStore ) {
+			stepIndex += numTotalSteps - this.state.numTotalSteps;
```

Comment:
I take it this is attempting to keep the progress from appearing to backtrack as missing plugin discovery increases the number of steps. I still saw it reverse during my testing, and this makes this component difficult to add more things to.  Maybe don't do the whole 3/6 business after all and just always have a step 1 for AT transfer (that is immediately "earned" if not an AT site) and always have a step 2 for AT initialization (that is immediately "earned" if not an AT site)

That should help keep this logic more extendable in the future, and less likely to suffer from weird math / backtrack issues.

Reply:
Yeah, exactly as you say.

> Maybe don't do the whole 3/6 business after all and just always have a step 1 for AT transfer (that is immediately "earned" if not an AT site) and always have a step 2 for AT initialization (that is immediately "earned" if not an AT site)

The problem is that we don't know how many steps there are in total before the site is not an Atomic one. So I could increase `stepIndex` to `1` or `2` for the AT process but out of how many steps? The progress bar would still backtrack on the `INITIALIZING` engineState.

I think I could do some proper math here with some constants (how much those steps roughly take in seconds) for the various steps.